### PR TITLE
Refactoring of the videostore

### PR DESCRIPTION
**Pull request by gregdel**
*  Remove slugs from everywhere
* Update the show index format to support path
* Add a HTTP route to render show details
* Move video, show and movie indexes to seperate packages 
* Rename functions to match the new package names
* Add a HTTP route to delete movies and episodes

### CODE REVIEW

File: app/internal/token/config_test.go [hunk 1]
```diff
@@ -33,7 +32,7 @@ var configFile = strings.NewReader(`
   include:
     - user
   allowed:
-    - DeleteBySlugs
+    - DeleteByID
```
Comment (PouuleT) [inline app/internal/token/config_test.go]: That route doesn't exist

File: app/internal/server/routes.go [hunk 2]
```diff
@@ -19,25 +19,20 @@ func (s *Server) httpServer(log *logrus.Entry) *http.Server {
...
+	mux.HandleFunc("/shows", s.showIds).Name("ShowsListIDs")
+	mux.HandleFunc("/shows/{id}", s.getShowDetails).Name("GetShowDetails")
+	mux.HandleFunc("/shows/{id}/{season:[0-9]+}/{episode:[0-9]+}", s.getShowEpisodeIDDetails).Name("GetShowEpisodeIDDetails")
 	mux.HandleFunc("/wishlist", s.wishlist).Name("Wishlist")
 
 	mux.HandleFunc("/torrents", s.addTorrent).Methods("POST").Name("TorrentsAdd")
 
 	if s.config.HTTPServer.ServeFiles {
 		log.Debug("server will be serving files")
-		mux.HandleFunc("/{videoType:movies|shows}/slugs/{slug}/delete", s.deleteFile).Name("DeleteBySlugs")
```
Comment (PouuleT) [inline app/internal/server/routes.go]: We should create a DeleteByID route

File: app/internal/token/config_test.go (hunk 1 above)
Comment (gregdel) [inline app/internal/token/config_test.go]: You're right

File: app/internal/server/routes.go (hunk 2 above)
Comment (gregdel) [inline app/internal/server/routes.go]: Let's create it ! :)

File: lib/library/library.go [hunk 3]
```diff
@@ -401,7 +401,8 @@ func (l *Library) DeleteShowEpisode(se *polochon.ShowEpisode, log *logrus.Entry)
 			return err
 		}
 		// Remove the season from the index
-		if err := l.showIndex.RemoveSeason(se.Show, se.Season, log); err != nil {
+		show := &polochon.Show{ImdbID: se.ShowImdbID}
```
Comment (PouuleT) [inline lib/library/library.go]: Why create a new show? Why not keep using se.Show ?

Comment (gregdel) [inline lib/library/library.go]: If you only have the ShowEpisode, the Show is nil and the whole thing panics...
This code needs to be refactored as well, to be refactored it needs tests... I' woking on it ;)

Comment (PouuleT) [inline lib/library/library.go]: Pretty fucked up if we can't be sure of what's inside our objects =)

**Pull request merged**